### ISSUE TITLE
literal_string_with_formatting_args false positive on test

### ISSUE THREAD

**Issue by tamird** [rust-lang/rust-clippy#13885]
### Summary

See reproducer below. I can't reproduce this in playground because it doesn't run clippy with `--all-targets`.

### Lint Name

literal_string_with_formatting_args

### Reproducer

I tried this code:

```rust
pub fn parse(_: &str) -> Result<Vec<String>, String> {
    unimplemented!()
}

#[test]
fn test_parse() {
    assert!(parse(
        #[allow(clippy::literal_string_with_formatting_args)]
        "foo {:}"
    )
    .is_err());
}
```

I saw this happen:

```shell
$ cargo +nightly clippy --all-targets
warning: this looks like a formatting argument but it is not part of a formatting macro
 --> src/lib.rs:5:6
  |
5 |     #[test]
  |      ^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#literal_string_with_formatting_args
  = note: `#[warn(clippy::literal_string_with_formatting_args)]` on by default
```

I expected to see this happen:
No warnings.

### Version

```text
rustc 1.85.0-nightly (917bfa784 2024-12-26)
binary: rustc
commit-hash: 917bfa78478cbcc77406e5ea37b24c3eedefacf4
commit-date: 2024-12-26
host: aarch64-apple-darwin
release: 1.85.0-nightly
LLVM version: 19.1.6
```

**Comment by DaniPopes** [comment]
Also just encountered this inside of `assert!()`. Similar/smaller:
```rust
pub fn bad() {
    let value = 0;
    assert!(format!("{value}").is_ascii());
}
```
```rust
warning: this looks like a formatting argument but it is not part of a formatting macro
 --> src/lib.rs:2:14
  |
2 |       let value = 0;
  |  ______________^
3 | |     assert!(format!("{value}").is_ascii());
  | |_^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#literal_string_with_formatting_args
  = note: `#[warn(clippy::literal_string_with_formatting_args)]` on by default
```

It looks like it tries to report the lint on a dummy span + some offset

**Comment by amircodota** [comment]
Got a few more examples

This is an actix-web route

```
#[route(
    "/forward/{path:.*}",
    method = "GET",
    method = "POST",
    method = "PUT",
    method = "OPTION"
)]
async fn forward(
    request: HttpRequest,
    body: Payload,
    data: Data<HttpState>,
    path: web::Path<String>,
) -> impl Responder {
```

This is error_chain bail macro

```
bail!("failed to prepare resumable config object: {e}")
```

In docs comment

```
/// Since there's no `lib.rs` or `main.rs` file here in the `tests` directory -
```

Also appears on some imports with the `with` word

```
use crate::common::server_utils::assert_notification_message_with_text;
```


```

**Comment by profetia** [comment]
Might be a FN on `from_expansion`

**Comment by samueltardieu** [comment]
Ping @GuillaumeGomez

**Comment by ilammy** [comment]
More examples of false positives.

Clippy doesn't like `.clippy.toml`!

    warning: this looks like a formatting argument but it is not part of a formatting macro
     --> [redacted]/.clippy.toml:2:27
      |
    2 | too-many-arguments-threshold = 9
      |                           ^^^
      |
      = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#literal_string_with_formatting_args
      = note: `#[warn(clippy::literal_string_with_formatting_args)]` on by default

I don't know whether it's the attribute or the import that's triggering it:

    warning: these look like formatting arguments but are not part of a formatting macro
     --> [redacted]/src/main.rs:1:22
      |
    1 | #![allow(clippy::too_long_first_doc_paragraph)]
      |                      ^^^^^^^^
    ...
    9 |         thread::{fresh_run, join_cluster, start_cluster},
      |               ^^^^^^^^
      |
      = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#literal_string_with_formatting_args
      = note: `#[warn(clippy::literal_string_with_formatting_args)]` on by default

**Comment by profetia** [comment]
Just find another way to trigger the FP

```rust
// file name: {value}.rs
#![crate_name = "strange_file_name"]
#![allow(unused)]

fn another_bad() {
    let value = 0;
    dbg!("something");
}

fn main() {}
```

**Comment by GuillaumeGomez** [comment]
Wow nice. Gonna take a look in the next days.

**Comment by GuillaumeGomez** [comment]
@profetia: I can't trigger the lint with your code.

**Comment by GuillaumeGomez** [comment]
I opened https://github.com/rust-lang/rust-clippy/pull/13953.

**Comment by profetia** [comment]
@GuillaumeGomez You need to put it in a file that is named as `{value}.rs`.

**Comment by ngoldbaum** [comment]
another false positive in PyO3, on `feature = "nightly"`.

https://github.com/PyO3/pyo3/actions/runs/12680326895/job/35341906091?pr=4845#step:7:1086

**Comment by larseggert** [comment]
I even see this hit on `.clippy.toml`, which is quite bizarre:
```error: this looks like a formatting argument but it is not part of a formatting macro
 --> /Users/lars/Documents/Code/neqo/.clippy.toml:2:68
  |
2 |     { path = "std::slice::from_raw_parts", reason = "see null_safe_slice" }
  |                                                                    ^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#literal_string_with_formatting_args
  = note: `-D clippy::literal-string-with-formatting-args` implied by `-D warnings`
  = help: to override `-D warnings` add `#[allow(clippy::literal_string_with_formatting_args)]`
```

**Comment by GuillaumeGomez** [comment]
Explanation in first comment of https://github.com/rust-lang/rust-clippy/pull/13953.

**Comment by amircodota** [comment]
Not sure if the fix for this is in nightly or not. Still on the latest nightly I get the following issues

1. warning on actix-web routes

```
#[route(
    "/forward/{path:.*}",
    method = "GET",
    method = "POST",
    method = "PUT",
    method = "OPTION"
)]
```

This is the warning generated

```
warning: this looks like a formatting argument but it is not part of a formatting macro
   --> shared/http_endpoints_server/src/server.rs:190:15
    |
190 |     "/forward/{path:.*}",
    |               ^^^^^^^^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#literal_string_with_formatting_args
    = note: `#[warn(clippy::literal_string_with_formatting_args)]` on by default
```

2. Warning on `error_chain::bail` macros

```
warning: this looks like a formatting argument but it is not part of a formatting macro
   --> deep/lunar_local/src/downloader_utils.rs:151:68
    |
151 |                     bail!("failed to initiate HTTPDownload Object: {e}")
    |                                                                    ^^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#literal_string_with_formatting_args
    = note: `#[warn(clippy::literal_string_with_formatting_args)]` on by default
```

3. warning on imports. Unclear why

```
use crate::common::template_rendering_utils::{
    call_show_notification_with, create_trigger_manager,
};
```

This is the warning

```
warning: this looks like a formatting argument but it is not part of a formatting macro
 --> tabnine/server/tests/template_rendering.rs:5:5
  |
5 |     call_show_notification_with, create_trigger_manager,
  |     ^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#literal_string_with_formatting_args
  = note: `#[warn(clippy::literal_string_with_formatting_args)]` on by default
```

Same here

```
use crate::common::server_utils::assert_notification_message_with_text;
```

```
warning: this looks like a formatting argument but it is not part of a formatting macro
 --> tabnine/server/tests/template_rendering.rs:3:60
  |
3 | use crate::common::server_utils::assert_notification_message_with_text;
  |                                                            ^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#literal_string_with_formatting_args
```

4. Warning in doc comments

```
/// This file must exist in order to have a package of tests here,
/// Since there's no `lib.rs` or `main.rs` file here in the `tests` directory -
/// This mod serves as a `mod.rs` to include the `advanced_completions_preview_tests` package in.
mod advanced_completions_preview_tests;
mod common;
```

This is the warning

```
warning: this looks like a formatting argument but it is not part of a formatting macro
 --> tabnine/server/tests/advanced_completions_tests_mod.rs:2:70
  |
2 |   /// Since there's no `lib.rs` or `main.rs` file here in the `tests` directory -
  |  ______________________________________________________________________^
3 | | /// This mod serves as a `mod.rs` to include the `advanced_completions_preview_tests` package in.
  | |_^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#literal_string_with_formatting_args
  = note: `#[warn(clippy::literal_string_with_formatting_args)]` on by default
```

**Comment by GuillaumeGomez** [comment]
I think some of the warnings might not be covered by https://github.com/rust-lang/rust-clippy/pull/13953. I'll check once merged.
